pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -222,3 +224,29 @@ func GetAllDeploymentNames(client mongodbatlas.Client, projectID string) ([]stri
 	}
 	return deploymentNames, nil
 }
+
+func normalizeInstanceSize(currentInstanceSize string, autoscaling *mdbv1.AdvancedAutoScalingSpec) string {
+	currentSize := extractNumberFromInstanceTypeName(currentInstanceSize)
+	minSize := extractNumberFromInstanceTypeName(autoscaling.Compute.MinInstanceSize)
+	maxSize := extractNumberFromInstanceTypeName(autoscaling.Compute.MaxInstanceSize)
+
+	if currentSize < minSize {
+		return autoscaling.Compute.MinInstanceSize
+	}
+
+	if currentSize > maxSize {
+		return autoscaling.Compute.MaxInstanceSize
+	}
+
+	return currentInstanceSize
+}
+
+func extractNumberFromInstanceTypeName(instanceTypeName string) int {
```

Comment:
It seems to me that this function is worth providing related context(comment, link to documentation).